contracts/fee/FeeHandler.sol

```diff
@@ -3,46 +3,215 @@
 pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
 import "../data/DataStore.sol";
 import "../role/RoleModule.sol";
+import "../utils/BasicMulticall.sol";
 import "../fee/FeeUtils.sol";
+import "../data/Keys.sol";
+import "../oracle/ChainlinkPriceFeedUtils.sol";
+import "../utils/Precision.sol";
+import "../v1/IVaultV1.sol";
+import "../v1/IVaultGovV1.sol";
 
 // @title FeeHandler
-contract FeeHandler is ReentrancyGuard, RoleModule {
+contract FeeHandler is ReentrancyGuard, RoleModule, BasicMulticall {
+    using SafeERC20 for IERC20;
+
+    struct FeeAmounts {
+        uint256 gmx;
+        uint256 wnt;
+    }
+
     DataStore public immutable dataStore;
     EventEmitter public immutable eventEmitter;
+    IVaultV1 public immutable vaultV1;
+    address public immutable gmx;
 
     constructor(
         RoleStore _roleStore,
         DataStore _dataStore,
-        EventEmitter _eventEmitter
+        EventEmitter _eventEmitter,
+        IVaultV1 _vaultV1,
+        address _gmx
     ) RoleModule(_roleStore) {
         dataStore = _dataStore;
         eventEmitter = _eventEmitter;
+        vaultV1 = _vaultV1;
+        gmx = _gmx;
     }
 
-    // @dev claim fees from the specified markets
-    // @param markets the markets to claim fees from
-    // @param tokens the fee tokens to claim
-    function claimFees(
-        address[] memory markets,
-        address[] memory tokens
-    ) external nonReentrant onlyFeeKeeper {
-        if (markets.length != tokens.length) {
-            revert Errors.InvalidClaimFeesInput(markets.length, tokens.length);
+    // @dev withdraw fees in buybackTokens from this contract
+    // @param marketTokens the markets from which to withdraw fees
+    // @param buybackToken the token for which to withdraw fees
+    function withdrawFees(address[] calldata markets, address buybackToken) external nonReentrant onlyFeeKeeper {
+        _validateBuybackToken(_getBatchSize(buybackToken), buybackToken);
+
+        for (uint256 i; i < markets.length; i++) {
+            FeeUtils.claimFees(dataStore, eventEmitter, markets[i], buybackToken, address(this));
         }
 
         address receiver = dataStore.getAddress(Keys.FEE_RECEIVER);
+        IERC20(buybackToken).safeTransfer(receiver, IERC20(buybackToken).balanceOf(address(this)));
+    }
+
+    // @dev claim fees in feeToken from the specified markets
+    // @param market the market from which to claim fees
+    // @param feeToken the fee tokens to claim from the market
+    function claimFees(address market, address feeToken) external nonReentrant {
+        if (_getBatchSize(feeToken) != 0) {
+            revert Errors.InvalidFeeToken(feeToken);
+        }
+
+        uint256 feeAmount;
+        if (market == address(0)) {
```

Comment:
i would suggest for `claimFees` func to accept `uint version` arg to make it more explicit and not rely on market

Reply:
and maybe create constants like `const uint V1 = 1`

Reply:
ok with me

Reply:
Ok, I've updated per G's recommendation above in my latest commit, including adding v1 and v2 const uint256 variables